x-pack/legacy/plugins/ml/server/models/results_service/results_service.js

```diff
@@ -402,11 +402,111 @@ export function resultsServiceProvider(callWithRequest) {
     return definition;
   }
 
+  /**
+   * Gets the record of partition filed - values pairs.
+   * @param criteriaFields
+   * @param earliestMs
+   * @param latestMs
+   * @returns {Promise<*>}
+   */
+  async function getPartitionFieldsValues(criteriaFields, earliestMs, latestMs) {
+    const resp = await callWithRequest('search', {
+      index: ML_RESULTS_INDEX_PATTERN,
+      size: 0,
+      body: {
+        query: {
+          bool: {
+            filter: [
+              ...criteriaFields.map(({ fieldName, fieldValue }) => {
+                return {
+                  term: {
+                    [fieldName]: fieldValue,
+                  },
+                };
+              }),
+              {
+                range: {
```

Comment:
As commented in the follow-up, should consider adding an extra term to the filter here, for `result_type`, depending on whether model plot has been enabled or not.